src/content/guides/packages.md

```diff
@@ -261,6 +261,174 @@ To update `pubspec.lock` run `dart pub get` without `--enforce-lockfile`.
 
 [content hash]: /tools/pub/glossary#content-hashes
 
+<a name="workspaces"></a>
+## Pub Workspaces (Mono-repo Support)
+
+When working on a project, you might develop multiple Dart packages in the same
+version control repository (a mono-repo).
+
+For example you might have a directory layout like: 
+
+```console
+/
+  packages/
+    shared/
+      pubspec.yaml
+      pubspec.lock
+      .dart_tool/package_config.json
+    client_package/
+      pubspec.yaml
+      pubspec.lock
+      .dart_tool/package_config.json
+    server_package/
+      pubspec.yaml
+      pubspec.lock
+      .dart_tool/package_config.json
+```
+
+There are multiple issues with this organization:
+
+* You need to run `pub get` once for each package.
+* You risk ending up with different versions of dependencies for each package.
+  Leading to confusion when context switching between the packages.
+* If you open the root folder in your IDE, the dart analyzer will have to have a
+  separate analysis contexts for each package.
+
+Pub allows you to organize your repository as a "workspace" using a single
+shared resolution for all your packages.
+
+To create a workspace:
```

Comment:
```suggestion
shared resolution for all your packages.

:::note
Using a single shared dependency resolution for all your packages increases
the risks of dependency conflicts. Because Dart does not allow multiple versions
of the same package.

If the packages are going to be used together (as is commonly the case),
this is often a feature, as it forces you to resolve incompatibilities between your
packages when the arise, rather then when you start using the packages.
:::

To create a workspace:
```